### DISCUSSION TITLE
Help with the PathStyleExtension

### DISCUSSION THREAD

**Discussion by kentstephen** [developmentseed/lonboard#526] [General]
Hi Kyle,

I searched for this extension on here and didn't see any discussions. I also did not find a specific example for this particular extension in the docs.

I have this snippet for a layer:

```python
from lonboard import PathLayer
from lonboard.layer_extension import PathStyleExtension
extension=PathStyleExtension()
b_layer = PathLayer(
    # Select only a few attribute columns from the table
    table = road_table.select(["geometry"]),
    get_color=[255, 255, 255],
    get_width=20,
    get_dash_array=[10, 5],
    extensions=[extension],
    dash_justified=True
)
```

I'm trying to make this layer zebra'd. When I isolate this layer I see the roads in white, but no dashes.

![image](https://github.com/developmentseed/lonboard/assets/43362477/83434b9f-18cd-49c1-bafc-c7ba758f108e)

I know ```get_dash_layer``` is required maybe I am missing something else.

**Comment by kylebarron** [comment]
Hmm. I didn't have an opportunity to actually test this code end to end before releasing (that's why there's no example). But I wanted to merge the contributed PR https://github.com/developmentseed/lonboard/pull/487. I don't see anything obviously wrong with that invocation at a glance.

**Reply by kentstephen** [reply]
I dont think I am going to end up using the extension for this map afterall, but I would be happy take screenshots and provide anything else you would need to look into it.

**Reply by kylebarron** [reply]
You need to pass `dash=True` into the constructor of `PathStyleExtension`,

```py
from lonboard.layer_extension import PathStyleExtension
extension = PathStyleExtension(dash=True)

layer = PathLayer.from_geopandas(
    gdf,
    width_min_pixels=0.8,
    get_dash_array=[10, 5],
    extensions=[extension],
    dash_justified=True
)
m = Map(layer)
m
```

<img width="643" alt="image" src="https://github.com/developmentseed/lonboard/assets/15164633/3520130a-490c-4a32-bc25-961976e254a1">

This [mirrors the upstream deck.gl API](https://deck.gl/docs/api-reference/extensions/path-style-extension#constructor) but is a bit unexpected. I'm not sure whether we should special case that. We should at least have better documentation for this
